user.php

```diff
@@ -217,4 +217,29 @@
 			exit();
 		}
 		break;
+	
+	/**
+	 *Added for fixing users who dont have permission and try to access a form leading to
+	 *a redirect loop.
+	 *Now it will jump back to main page and show a message indicating an error msg.
+	 *
+	 *other functions can be added to this part accordingly
+	 *
+	 *Added By Jinfu Feb 2015
+	 */
+	case 'nopermission':
+		if(!icms::$user){
+			redirect_header('index.php',5,_US_NOPERMISS);
+		} elseif (!empty($_GET['xoops_redirect'])) {
+			$redirect = htmlspecialchars(trim($_GET['xoops_redirect']));
+			$isExternal = FALSE;
+			$tragetURL=substr(ICMS_URL,0,strpos(ICMS_URL,"/formulize")).$redirect;
+			redirect_header(ICMS_URL."/modules/formulize/application.php?id=all",5, _US_NOPERMISS."<br />Viewing: '".$tragetURL."'<br />");

```

Comment:
The error message would be better if it said, "Sorry, you do not have permission to view this page. Please contact your site administrator." Adding the link to the page is not necessary.